src/native/eventpipe/ep-session.c

```diff
@@ -523,6 +620,246 @@ ep_session_write_all_buffers_to_file (EventPipeSession *session, bool *events_wr
 	return !ep_file_has_errors (session->file);
 }
 
+static
+void
+session_disable_user_events (EventPipeSession *session)
+{
+	EP_ASSERT (session != NULL);
+	ep_return_void_if_nok (session->session_type == EP_SESSION_TYPE_USEREVENTS && session->user_events_data_fd > 0);
+
+	ep_requires_lock_held ();
+
+	EventPipeSessionProviderList *providers = session->providers;
+	EP_ASSERT (providers != NULL);
+	DN_LIST_FOREACH_BEGIN (EventPipeSessionProvider *, session_provider, ep_session_provider_list_get_providers (providers)) {
+		ep_session_provider_unregister_tracepoints (session_provider, session->user_events_data_fd);
+	} DN_LIST_FOREACH_END;
+
+#if HAVE_UNISTD_H
+	close (session->user_events_data_fd);
+#endif // HAVE_UNISTD_H
+	session->user_events_data_fd = 0;
```

Comment:
0 is actually a valid file descriptor so we should probably use -1 instead that means an invalid file descriptor.

Reply:
Changed